files/external_node_v2.rb

```diff
@@ -377,6 +382,12 @@ def watch_and_send_facts(parallel)
             upload_facts(certname, req)
           end
 
+          if push_only_current_client_facts
+            require 'yaml'
+            empty_env_result = { "classes" => {}}
+            puts empty_env_result.to_yaml
```

Comment:
This isn't quite correct. If the fact upload goes wrong for whatever reason, you'll be served the cache if available. That means switching from ENC to non-ENC could give some weird results. I think you should do:
```ruby
if push_only_current_client_facts
  require 'yaml'
  empty_env_result = { "classes" => {}}
  result = empty_env_result.to_yaml
 else
  result = enc(certname)
  cache(certname, result)
end
```

Reply:
If I understand you correctly, if the fact upload goes wrong, we should end up in the `rescue` part (https://github.com/theforeman/puppet-foreman/pull/670/files#diff-56980159eb7873d008e28343dd19c1f3L383), which means that the `exit 0` gets skipped.

So I would need to fix the `rescue` part by adding the empty yaml puts _and_ `exit 0` to it if `push_only_current_client_facts` is set.

With your suggested snippet the `result = read_cache(certname)` and the puts of it will still happen, because the `exit 0` was skipped.

Reply:
You're right. I did mean that's the problem but my solution is incorrect. I would prefer to setting `result = ...` rather than exiting early. Looking at the code I think it could use some more helper functions but that doesn't have to be part of this PR.

Reply:
Ah, I understand. I've tried to avoid the premature `exit` here: https://github.com/theforeman/puppet-foreman/pull/670/commits/aa7638cea2dd3980be542d310c57a3dd1a2a6265

I'll rebase later and I'm also okay with renaming the param to `--no-env` although I prefer more descriptive arguments in general :smirk: